sdk/appcenter-crashes/src/test/java/com/microsoft/appcenter/crashes/CrashesTest.java

```diff
@@ -1647,4 +1657,119 @@ public boolean matches(Object argument) {
         verifyStatic();
         FileManager.write(any(File.class), eq(jsonCrash));
     }
+
+    @Test
+    public void handlerMemoryWarning() {
+
+        /* Mock classes. */
+        final Context mockContext = mock(Context.class);
+        ArgumentCaptor<ComponentCallbacks2> componentCallbacks2Captor = new ArgumentCaptor<>();
+        doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                return null;
+            }
+        }).when(mockContext).registerComponentCallbacks(componentCallbacks2Captor.capture());
+
+        /* Instance crash module. */
+        Crashes crashes = Crashes.getInstance();
+        crashes.onStarted(mockContext, mock(Channel.class), "", null, true);
+        crashes.applyEnabledState(true);
+        componentCallbacks2Captor.getValue().onConfigurationChanged(mock(Configuration.class));
+
+        /* Invoke callback onTrimMemory. */
+        componentCallbacks2Captor.getValue().onTrimMemory(TRIM_MEMORY_RUNNING_CRITICAL);
+
+        /* Verify put data to preferences. */
+        verifyStatic();
+        SharedPreferencesManager.putInt(eq(PREF_KEY_MEMORY_RUNNING_LEVEL), eq(TRIM_MEMORY_RUNNING_CRITICAL));
+
+        /* Invoke callback onLowMemory. */
+        componentCallbacks2Captor.getValue().onLowMemory();
+
+        /* Verify put data to preferences. */
+        verifyStatic();
+        SharedPreferencesManager.putInt(eq(PREF_KEY_MEMORY_RUNNING_LEVEL), eq(TRIM_MEMORY_COMPLETE));
+    }
+
+    @Test
+    public void registerAndUnregisterComponentCallbacks() {
+
+        /* Mock classes. */
+        final Context mockContext = mock(Context.class);
+        mockStatic(ErrorLogHelper.class);
+        when(ErrorLogHelper.getErrorStorageDirectory()).thenReturn(errorStorageDirectory.getRoot());
+        when(ErrorLogHelper.getStoredErrorLogFiles()).thenReturn(new File[0]);
+        when(ErrorLogHelper.getNewMinidumpFiles()).thenReturn(new File[0]);
+
+        /* Instance crash module. */
+        Crashes crashes = Crashes.getInstance();
+        crashes.onStarting(mAppCenterHandler);
+        crashes.setInstanceEnabled(true);
+        crashes.onStarted(mockContext, mock(Channel.class), "", null, true);
+        verifyStatic();
+        SharedPreferencesManager.remove(eq(PREF_KEY_MEMORY_RUNNING_LEVEL));
+
+        /* Verify register callback. */
+        verify(mockContext).registerComponentCallbacks(any(ComponentCallbacks2.class));
+
+        /* Disable crashes. */
+        crashes.applyEnabledState(false);
+
+        /* Verify unregister callback. */
+        verify(mockContext).unregisterComponentCallbacks(any(ComponentCallbacks2.class));
+
+        /* Verify clear preferences. */
+        verifyStatic(times(2));
+        SharedPreferencesManager.remove(eq(PREF_KEY_MEMORY_RUNNING_LEVEL));
+    }
+
+    @Test
+    public void setReceiveMemoryWarningInLastSession() {
```

Comment:
consider separate methods with share private method, this is very hard to read the end of the test